contracts/raffle/src/lib.rs

```diff
@@ -112,6 +112,19 @@ impl RaffleContract {
         storage.set(&DataKey::AlreadyInitialized, &true);
     }
 
+
+    /// A 'dummy' method that needs to be called by user before buying the ticket.
+    /// This a workaround to this issue https://github.com/eigerco/nebula/issues/41
+    /// 
+    /// # Arguments
+    ///
+    /// - `_` - The environment for this contract - not used here
+    /// - `by` - The address that is registering.
+    pub fn register(_: Env, by: Address) {
```

Comment:
Do we still need this @mariopil?

Reply:
Yes, that's still needed.

Reply:
I also suffered this in the governance contract. I had to implement the same. Very weird. And looks like as @mariopil [previously asked](https://discordapp.com/channels/897514728459468821/1157276028327166024/1157276028327166024) its something the community is aware of. So IMO, we should continue and maybe, add a specific contract in our catalog that checks this fail is still there, so whenever its fixed, we will be notified. Would this be a plausible option for now ?

Reply:
Just a simple integration test change is needed to see whether it's working or not. I don't think anything other is necessary. Let's leave it as it is and maybe add an issue so that we don't forget about it and run tests once in a while to see whether they've fixed the issue or not.

Reply:
Why don't we do a method called `get_raffle_state` which can tell you the state of the raffle? You can call this method instead of register, and this method would be usable in other scenarios. WDYT?

Reply:
What that method would return?

Reply:
You can have data from these keys:
```
enum DataKey {
    Admin = 1,
    Candidates = 2,
    MaxWinnerCount = 3,
    TicketPrice = 4,
    Token = 5,
    AlreadyInitialized = 6,
    AlreadyPlayed = 7,
}
```
How about the number of candidates, ticket_price, if already played?

Reply:
Yes, I think it's better to have something more explicit.

Reply:
It is not mandatory. In the marketplace contract I use get_listing but its not mandatory. I dont have a register method and the tests work well.

Reply:
Yes, in the tests you are calling `get_listing` before `buy_listing` on the same user that's why tests work. If you would call to `buy_listing` first you would have the same error. The problem is only with methods that use token.

Reply:
I was suggesting `get_raffle_state` as  such a work around. But all in all this problem is on Stellar's network.
If we were to merge, Does this mean to use the contract I have to call `init` and `register`?

Reply:
Still it means I cant just buy a ticket, I have to call register first if I am a user who is new?
I think we may need to add that to docs or something?

Reply:
I'll update the docs.